container/kubernetes/onboarding/kubernetes_deployment_creation.sh

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2020, Intel Corporation
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#    * Redistributions of source code must retain the above copyright notice,
+#      this list of conditions and the following disclaimer.
+#    * Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the following disclaimer in the
+#      documentation and/or other materials provided with the distribution.
+#    * Neither the name of Intel Corporation nor the names of its contributors
+#      may be used to endorse or promote products derived from this software
+#      without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+## Gateway IOT Device Deployments Pods Creation,Update and Delete.
+
+NODENAME=${1}
+APPNAME=${2}
+NAMESPACE=${NAMESPACE:-${3}}
+DEPLOYTYPE=${4}
+CONFIG_MAP_NAME=app-config
+GATEWAYAPPCONFIGTARGET=application.yaml
+DATASERVICECONFIGTARGET=fusiondataservice_application.yaml
+if [ "$#" -gt 4 -o "$#" -lt 4 -o "$#" -eq 3 ]; then
+	echo usage: ${0##*/} \<node-name\> \<application-name\> \<namespace\> \<deploytype\>
+	echo e.g. ${0##*/} kubemaster testsensor
+	exit 1
+fi
+
+
+cd $APPNAME
+TEST_SENSOR_CONFIG_EXIST=0
+TEST_SENSOR_CONFIG_EXIST=`grep "name: testsensor-config" deployment.yaml | wc -l`
+###### Find out correct DATASERVICECONFIG FROM Deployment yaml file
+if [ $TEST_SENSOR_CONFIG_EXIST -eq 0 ] 
+then
+	DATACONFIGTARGET=`grep "fusion*.*yaml" deployment.yaml | awk '{print $3}'`
+	mv $DATASERVICECONFIGTARGET $DATACONFIGTARGET
+	DATASERVICECONFIGTARGET=$DATACONFIGTARGET
+fi
+echo Creating temporary subdir for processing...
+WORKDIR=`mktemp -d -p "."`
+
+if [ $DEPLOYTYPE = "update" ]
```

Comment:
this if [ ] will throw an error if $DEPLOYTYPE is empty. please put it in quotest, i.e.  [ "$DEPLOYTYPE" = "update" ]